p2p/transport/webrtc/stream_write.go

```diff
@@ -0,0 +1,204 @@
+package libp2pwebrtc
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p/p2p/transport/webrtc/internal/async"
+	pb "github.com/libp2p/go-libp2p/p2p/transport/webrtc/pb"
+	"github.com/libp2p/go-msgio/pbio"
+	"github.com/pion/datachannel"
+	"google.golang.org/protobuf/proto"
+)
+
+// Package pion detached data channel into a net.Conn
+// and then a network.MuxedStream
+type (
+	webRTCStreamWriter struct {
+		stream *webRTCStream
+
+		writer    pbio.Writer
+		writerMux sync.Mutex
+
+		deadline    time.Time
+		deadlineMux sync.Mutex
+
+		deadlineUpdated async.CondVar
+		writeAvailable  async.CondVar
+
+		readLoopOnce sync.Once
+		closeOnce    sync.Once
+	}
+)
+
+func (w *webRTCStreamWriter) Write(b []byte) (int, error) {
+	w.stream.wg.Add(1)
+	defer w.stream.wg.Done()
```

Comment:
What is this waitgroup supposed to do?

Reply:
Ensure we do not close while we are writing or reading?

Reply:
Removed it, as I suppose the contract of the transport users makes it that this is not required, I can make more assumptions then I imagined